app/javascript/admin/controllers/creator_settings_controller.js

```diff
@@ -70,13 +71,29 @@ export class CreatorSettingsController extends Controller {
   }
 
   /**
-   * Updates ths branding/colors on the Creator Settings Page.
-   *
+   * Validates the color contrast for accessibility,
+   * if the contrast is okay, it updates the branding,
+   * else it displays the error.
    * @param {Event} event
    */
-  updateBranding(event) {
+  handleValidationsAndUpdates(event) {
     const { value: color } = event.target;
 
+    if (isLowContrast(color)) {
+      this.colorContrastErrorTarget.classList.remove('hidden');
```

Comment:
I think we need to consider some way to notify screen reader users that this error now applies too.

One option would be to add an `aria-live` to the `div` containing the message, and only populate the text here (which would mean we don't have to hide it too). For example:

```html
<div data-creator-settings-target="colorContrastError" aria-live="polite" class="mt-1 color-accent-danger">
    The selected color must be darker for accessibility purposes.
</div>
```
```js
if (isLowContrast(color)) {
  this.colorContrastErrorTarget.innerText = "The selected color must be darker for accessibility purposes.";
} else {
  this.updateBranding(color);
  this.colorContrastErrorTarget.innerText = "";
}
```

Reply:
✅